scripts/west_commands/sign.py

```diff
@@ -467,7 +467,12 @@ def sign(self, command, build_dir, build_conf, formats):
             else:
                 log.die(msg)
 
-        kernel_name = build_conf.get('CONFIG_KERNEL_BIN_NAME', 'zephyr')
+        sign_config_extra_args = config_get_words(command.config, 'rimage.extra-args', [])
+
+        if '-r' in sign_config_extra_args + args.tool_args:
+            kernel_name = 'llext'
+        else:
+            kernel_name = build_conf.get('CONFIG_KERNEL_BIN_NAME', 'zephyr')
```

Comment:
Does the `bootloader` code below still apply to this new `llext` thing?

Reply:
Sorry I missed that. Then this would be clearer and better (assuming peeking at `-r` is a good idea in the first place:

```python
is_llext = '-r' in sign_config_extra_args + args.tool_args

kernel_name = 'llext' if is_llext else ...

...

if is_llext:

```

Reply:
BTW CONFIG_KERNEL_BIN_NAME used to be hardcoded to `zephyr.elf` in this script. In #59721 such hardcoding was deemed a bad idea removing. Hardcoding something again to `llext` seems equally bad (but hard to tell for sure without a complete use case).